ingest.py

```diff
@@ -91,49 +93,63 @@ def load_single_document(file_path: str) -> Document:
-            glob.glob(os.path.join(source_dir, f"**/*{ext}"), recursive=True)
-        )
-    filtered_files = [file_path for file_path in all_files if file_path not in ignored_files]
+    if ignored_files is None:
+        ignored_files : List[str] = []
+
+    filtered_files = []
+
+    ext_pattern = re.compile(
+        r"({})$".format("|".join(re.escape(ext) for ext in LOADER_MAPPING)),
+        re.IGNORECASE
+    )
+
+    for root, _, files in os.walk(source_dir):
+        for file in files:
+            if ext_pattern.search(file):
+                file_path = os.path.join(root, file)
+                if file_path not in ignored_files:
+                    filtered_files.append(file_path)
 
     with Pool(processes=os.cpu_count()) as pool:
         results = []
         with tqdm(total=len(filtered_files), desc='Loading new documents', ncols=80) as pbar:
-            for i, doc in enumerate(pool.imap_unordered(load_single_document, filtered_files)):
+            for _, doc in enumerate(pool.imap_unordered(load_single_document, filtered_files)):
                 results.append(doc)
                 pbar.update()
 
     return results
 
-def process_documents(ignored_files: List[str] = []) -> List[Document]:
+def process_documents(ignored_files: Optional[List[str]] = None) -> List[Document]:
     """
     Load documents and split in chunks
     """
+
+    if ignored_files is None:
+        ignored_files : List[str] = []
+
     print(f"Loading documents from {source_directory}")
     documents = load_documents(source_directory, ignored_files)
     if not documents:
         print("No new documents to load")
-        exit(0)
+        sys.exit(0)
     print(f"Loaded {len(documents)} new documents from {source_directory}")
     text_splitter = RecursiveCharacterTextSplitter(chunk_size=chunk_size, chunk_overlap=chunk_overlap)
     texts = text_splitter.split_documents(documents)
     print(f"Split into {len(texts)} chunks of text (max. {chunk_size} tokens each)")
     return texts
 
-def does_vectorstore_exist(persist_directory: str) -> bool:
+def does_vectorstore_exist(persist_path: str) -> bool:
     """
     Checks if vectorstore exists
     """
-    if os.path.exists(os.path.join(persist_directory, 'index')):
-        if os.path.exists(os.path.join(persist_directory, 'chroma-collections.parquet')) and os.path.exists(os.path.join(persist_directory, 'chroma-embeddings.parquet')):
-            list_index_files = glob.glob(os.path.join(persist_directory, 'index/*.bin'))
-            list_index_files += glob.glob(os.path.join(persist_directory, 'index/*.pkl'))
+    if os.path.exists(os.path.join(persist_path, 'index')):
+        if os.path.exists(os.path.join(persist_path, 'chroma-collections.parquet')) and os.path.exists(os.path.join(persist_path, 'chroma-embeddings.parquet')):
+            list_index_files = glob.glob(os.path.join(persist_path, 'index/*.bin'))
+            list_index_files += glob.glob(os.path.join(persist_path, 'index/*.pkl'))
```

Comment:
I guess we could use `walk` here too and remove `glob` entirely, or `listdir`.

Reply:
@PulpCattel Optimized in the update.